iphone/Maps/UI/PlacePage/Components/OpenInAppActionSheet/OpenInApplication.swift

```diff
@@ -0,0 +1,123 @@
+enum OpenInApplication: Int, CaseIterable {
+  case osm
+  case googleMaps
+  case appleMaps
+  case osmAnd
+  case yandexMaps
+  case dGis
+  case cityMapper
+  case moovit
+  case uber
+  case waze
+}
+
+extension OpenInApplication {
```

Comment:
The key point of my solution is:
1. the data associated with the target application should be defined only the one time because its impossible situation that there can be >1 same application. This is why enum is better approach -  it isolates creational and usage process
2. every app always have the same set of options

You approach with the structs pushses to create a separate type for the every new application.

Reply:
My main concern was that you could miss adding a switch case statement when you introduce a new case for the enum. I've just realized that this produces a compiler error in swift, so your solution is fine.

I'm used to enums in Java but they are apparently different to enums in swift.